renderdoc/driver/vulkan/vk_shader_cache.cpp

```diff
@@ -1017,11 +1017,28 @@ void VulkanShaderCache::MakeGraphicsPipelineInfo(VkGraphicsPipelineCreateInfo &p
     rs.pNext = &provokeSetup;
   }
 
+  static VkPipelineCreateFlags2CreateInfoKHR flags2 = {
+      VK_STRUCTURE_TYPE_PIPELINE_CREATE_FLAGS_2_CREATE_INFO_KHR,
+  };
+
+  VkPipelineCreateFlags2KHR flags = pipeInfo.flags;
+
   // never create derivatives
-  ret.flags &= ~VK_PIPELINE_CREATE_DERIVATIVE_BIT;
+  flags &= ~VK_PIPELINE_CREATE_DERIVATIVE_BIT;
 
-  ret.flags &= ~VK_PIPELINE_CREATE_LIBRARY_BIT_KHR;
-  ret.flags &= ~VK_PIPELINE_CREATE_RETAIN_LINK_TIME_OPTIMIZATION_INFO_BIT_EXT;
+  flags &= ~VK_PIPELINE_CREATE_LIBRARY_BIT_KHR;
+  flags &= ~VK_PIPELINE_CREATE_RETAIN_LINK_TIME_OPTIMIZATION_INFO_BIT_EXT;
+
+  if(flags >= 0x80000000u)
```

Comment:
It would be nice to explicitly check `flags > VK_PIPELINE_CREATE_FLAG_BITS_MAX_ENUM` here to better document what this check is for (e.g. at first I thought this was wrong until I checked and saw that vulkan doesn't use the top bit of enums - I think to avoid signed/unsigned issues).